File: 1/1000s/19s/1910/RemoveAllOccurrencesOfASubstring.java
/*
https://leetcode.com/problems/remove-all-occurrences-of-a-substring/description/?envType=daily-question&envId=2025-02-11
*/

class Solution {
    public String removeOccurrences(String s, String part) {
        StringBuilder temporaryS=new StringBuilder(s);
        while(true){
            int lengthOfS=temporaryS.length();
            if(lengthOfS==0) break;
            int firstIndexOfPart=(String.valueOf(temporaryS)).indexOf(part);
            if(firstIndexOfPart!=-1){
                int count=part.length();
                while(count-- >0){
                    temporaryS.setCharAt(firstIndexOfPart++, '#');
                }
                temporaryS=new StringBuilder(String.valueOf(temporaryS).replaceAll("#", ""));
            } else break;
        }
        return String.valueOf(temporaryS);
    }
}

File: 1/1000s/19s/1910/RemoveAllOccurrencesOfASubstring.cpp
/*
https://leetcode.com/problems/remove-all-occurrences-of-a-substring/description/?envType=daily-question&envId=2025-02-11
*/

class Solution {
public:
    string removeOccurrences(string s, string part) {
        while(true){
            int sz=s.size();
            if(sz==0) break;
            auto itr=s.find(part);
            if(itr!=string::npos){
                int bgnidx=(int)itr, count=part.length();
                while(count--){
                    s[bgnidx++]='#';
                }
                s.erase(remove(s.begin(), s.end(), '#'), s.end());
            } else break;
        }
        return s;
    }
};